backend/shelf-function/src/main/java/com/htec/shelffunction/service/FunctionService.java

```diff
@@ -175,4 +178,77 @@ public void deleteFunction(Long functionId) {
 
         functionRepository.delete(functionEntity);
     }
+
+    public void createCustomFunction(CustomFunctionRequestModel customFunctionRequestModel, Long userId) {
+
+        FunctionEntity functionEntity = FunctionMapper.INSTANCE
+                .customFunctionRequestModelToFunctionEntity(customFunctionRequestModel);
+
+        functionEntity.setLanguage(customFunctionRequestModel.getLanguage());
+        functionEntity.setCustom(true);
+
+        functionRepository.saveAndFlush(functionEntity);
+
+        functionEntity.setPath(userId + pathSeparator + "functions" + pathSeparator + "Function" + functionEntity.getId());
+
+        functionRepository.save(functionEntity);
+
+        compileCustomFunction(functionEntity.getId(), customFunctionRequestModel, userId);
+    }
+
+    public void compileCustomFunction(Long functionEntityId, CustomFunctionRequestModel customFunctionRequestModel, Long userId) {
+
+        try {
+            String extension = "";
+            String compileCommand = "";
+
+            if (Objects.equals(customFunctionRequestModel.getLanguage(), "java")) {
+                extension += JAVA_EXTENSION;
+            } else {
+                extension += CSHARP_EXTENSION;
+            }
+
+            String sourceFilePath = homePath +
+                    userPath +
+                    PREDEFINED_FUNCTION_FOLDER +
+                    pathSeparator +
+                    "hello" +
+                    extension;
+
+            String sourceFileContent;
```

Comment:
why declaration and assigning values in two line

Reply:
Changed